### PR TITLE
Ignore contenteditable elements in no-noninteractive-element-interactions

### LINKED ISSUE #525: Should contentEditable elems be excempt from no-noninteractive-element-interactions?
I have a `pre` element set as `contentEditable` for reasons. Thoughts on removing `contentEditable` elements from the [no-noninteractive-element-interactions](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-element-interactions.md) rule?

I also added `role="textbox"` since that's really what it is, but I have other `input`s inserting content in specific areas of the `contentEditable` elem; this is why it's _not_ an actual `textarea`. However, doing so comes back with the [no-noninteractive-element-to-interactive-role](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-element-to-interactive-role.md) error.

Here's the snippet in question:

```html
<pre
  aria-label="Edit description"
  aria-multiline="true"
  contentEditable
  onKeyUp={this.doSomething}
  role="textbox"
>
  <!-- 
    Other HTML which builds the content of this container
    which, in turn, can also be edited directly.
  -->
</pre>
```

My thoughts are since this has `contentEditable` it's now an interactive element and could be exempt from this rule.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Fixes #525

### CODE REVIEW

File: src/util/isContentEditable.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { getProp } from "jsx-ast-utils";
+
+const isContentEditable = (tagName, attributes) => {
+  const prop = getProp(attributes, "contentEditable");
+  return prop && prop.value?.raw !== '"false"';
```
Comment (ljharb) [inline src/util/isContentEditable.js:5]:
```suggestion
  return getProp(attributes, "contentEditable")?.value?.raw !== '"false"';
```
?

**Review by ljharb — APPROVED** [review]
LGTM

File: src/util/isContentEditable.js (hunk 1 above)
Comment (JoshuaKGoldberg) [reply src/util/isContentEditable.js:5]: 👌 snazzy

File: src/util/isContentEditable.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { getProp } from "jsx-ast-utils";
+
+const isContentEditable = (tagName, attributes) => {
+  return getProp(attributes, "contentEditable")?.value?.raw !== '"false"';
+};
+
+export default isContentEditable;
```
Comment (ljharb) [inline src/util/isContentEditable.js:7]:
```suggestion
export default function isContentEditable(tagName, attributes) {
  return getProp(attributes, 'contentEditable')?.value?.raw !== '"false"';
};
```

Comment (JoshuaKGoldberg) [reply src/util/isContentEditable.js]: Love it -- I was just going off the existing sibling util files. +1

File: src/util/isContentEditable.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { getProp } from 'jsx-ast-utils';
+
+export default function isContentEditable(tagName, attributes) {
+  return getProp(attributes, 'contentEditable')?.value?.raw !== '"false"';
```
Comment (JoshuaKGoldberg) [inline src/util/isContentEditable.js:4]: Hmm, actually, what if it's `undefined` for not existing... will update.

File: src/util/isContentEditable.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { getProp } from 'jsx-ast-utils';
+
+export default function isContentEditable(tagName, attributes) {
```
Comment (ljharb) [inline src/util/isContentEditable.js:3]: this one needs tests (i'll add them shortly)

**Comment by ljharb** [comment]
@JoshuaKGoldberg so, i rebased this, and tweaked the logic slightly (from "not false" to "is true") based on my testing - iow, that browsers normalize all values outside the enum of "true", "false", or "inherit" to "false" (and the JS setter refuses any value but those three), and now some tests are failing.

Mind resetting your local branch, and taking a look?

File: src/util/isContentEditable.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  const prop = getProp(attributes, 'contentEditable');
+
+  if (!prop || !prop.value || typeof prop.value.raw !== 'string') {
```
Comment (JoshuaKGoldberg) [inline src/util/isContentEditable.js:6]:
```suggestion
  if (!prop?.value || typeof prop.value.raw !== 'string') {
```

File: src/util/isContentEditable.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    return false;
+  }
+
+  return prop.value.raw === '"true"';
```
Comment (JoshuaKGoldberg) [inline src/util/isContentEditable.js:10]: Hmm, is this a result of the way JSX attributes get parsed? In my Chrome 104, `<div contenteditable>` works the same as `<div contentEditable="true">`.

Screenshot of adding contenteditable to a div in Chrome dev tools, and the div being contenteditable

Comment (ljharb) [reply src/util/isContentEditable.js:10]: yes, if you check out that HTML element and look at `.contentEditable` you'll get the string true.

File: __mocks__/JSXAttributeMock.js [hunk 7]
```diff
@@ -24,6 +25,8 @@ export default function JSXAttributeMock(prop: string, value: mixed, isExpressio
   let attributeValue = astValue;
   if (isExpressionContainer || astValue.type !== 'Literal') {
     attributeValue = JSXExpressionContainerMock(astValue);
+  } else if (attributeValue.type === 'Literal' && !('raw' in attributeValue)) {
+    attributeValue.raw = JSON.stringify(attributeValue.value);
```
Comment (JoshuaKGoldberg) [inline __mocks__/JSXAttributeMock.js:29]: Sorry, I am a Flow noob and it exits with `Unknown system error -86` locally. I don't know how to fix the type errors here. 🙃 

...can we switch to TypeScript? 😄

Comment (ljharb) [reply __mocks__/JSXAttributeMock.js:29]: I'll take a look at it.

Separately, flow's been dead for years, but a) i'm not a fan of TS, and b) the work involved is likely not small.

**Comment by JoshuaKGoldberg** [comment]
I'm working on the Flow issues now, will rebase once I'm done...

**Pull request merged**